ssl/tls_srp.c

```diff
@@ -233,6 +233,8 @@ int SSL_set_srp_server_param_pw(SSL *s, const char *user, const char *pass,
         return -1;
     s->srp_ctx.N = BN_dup(GN->N);
     s->srp_ctx.g = BN_dup(GN->g);
+    if (s->srp_ctx.N == NULL || s->srp_ctx.g == NULL)
+        return -1;
```

Comment:
This introduced a memory leak in case the first one succeeds and the second one fails.

Reply:
Or maybe not, it might get cleaned up as part of cleaning up s.

Reply:
Since there is no memory free if 'SRP_create_verifier_BN_ex' fails,  I think it is unnecessary to release the memory if 'BN_dup' fails.

Reply:
Another way to fix this is to pass these values to SRP_create_verifier_BN_ex() which checks for NULL N and g.

Reply:
Fine, I have submitted a fixup.